### PR TITLE
fix recovery from provision clients error

### PR DESCRIPTION
**Pull request by kubamahnert**

### CODE REVIEW

File: lib/gooddata/lcm/lcm2.rb [hunk 1]
```diff
@@ -365,6 +359,17 @@ def perform(mode, params = {})
         }
       end
 
+      def run_action(action, params)
+        GoodData.logger.info("Running #{action.name} action ...")
+        params.clear_filters
+        # Check if all required parameters were passed
+        BaseAction.check_params(action.const_get('PARAMS'), params)
```
Comment (yenkins) [inline lib/gooddata/lcm/lcm2.rb]: Calls 'action.const_get('params')' 2 times (DuplicateMethodCall)

File: lib/gooddata/lcm/lcm2.rb [hunk 2]
```diff
@@ -365,6 +359,17 @@ def perform(mode, params = {})
         }
       end
 
+      def run_action(action, params)
+        GoodData.logger.info("Running #{action.name} action ...")
+        params.clear_filters
```
Comment (yenkins) [inline lib/gooddata/lcm/lcm2.rb]: Calls 'params.clear_filters' 2 times (DuplicateMethodCall)

File: lib/gooddata/lcm/lcm2.rb [hunk 3]
```diff
@@ -365,6 +359,17 @@ def perform(mode, params = {})
         }
       end
 
+      def run_action(action, params)
```
Comment (yenkins) [inline lib/gooddata/lcm/lcm2.rb]: Has approx 7 statements (TooManyStatements)

File: lib/gooddata/lcm/lcm2.rb [hunk 4]
```diff
@@ -365,6 +359,17 @@ def perform(mode, params = {})
...
+        GoodData.logger.info("Running #{action.name} action ...")
+        params.clear_filters
+        # Check if all required parameters were passed
+        BaseAction.check_params(action.const_get('PARAMS'), params)
+        params.setup_filters(action.const_get('PARAMS'))
+        out = action.send(:call, params)
+        params.clear_filters
+        return out
```
Comment (yenkins) [inline lib/gooddata/lcm/lcm2.rb]: Style/RedundantReturn: Redundant `return` detected.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/921/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/771/ : FAILURE in 2m 37s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1039/ : SUCCESS in 1m 53s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/924/ : SUCCESS in 21s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/775/ : SUCCESS in 31s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1043/ : SUCCESS in 39s

**Pull request merged**